src/schema/sale/end_sale_mutation.js

```diff
@@ -0,0 +1,43 @@
+// @ts-check
+
+import { GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { SaleType } from "schema/sale/index"
+
+export default mutationWithClientMutationId({
+  name: "EndSale",
+  description:
+    "Mark sale as ended.",
+  inputFields: {
+    sale_id: {
+      type: GraphQLString,
+    },
+  },
+  outputFields: {
+    sale: {
+      type: SaleType,
+      resolve: (
+        { sale_id },
+        _,
+        _request,
+        { rootValue: { saleLoader } }
+      ) => saleLoader(sale_id),
```

Comment:
#minor but rather than returning just `sale_id` and fetching the sale, the result of calling the Gravity API endpoint returns the sale, so you can save this call and just resolve with the data returned from the endpoint.

So this could be the simple `resolve: sale => sale`.